source/m3_module.c

```diff
@@ -27,13 +32,27 @@ void  m3_FreeModule  (IM3Module i_module)
     }
 }
 
+static void Module_FreeFunctions(IM3Module i_module)
+{
+    for (u32 i = 0; i < i_module->numFunctions; ++i)
+    {
+        IM3Function func = &i_module->functions[i];
+        m3Free (func->constants);
+        if (func->name != func->import.fieldUtf8)
+        {
+            m3Free (func->name);
```

Comment:
This particularly feels like a hack; another option i see is to introduce a flag (`func->nameAllocated`) and set it whenever `name` needs to be freed.